server/cp/ai/semaphore.py

```diff
@@ -368,50 +369,65 @@ def create_tag_in_semaphore(self, html_content: Dict[str, str]) -> Dict:
             logger.error(
                 f"Semaphore Create Tag Failed failed. We are in analyze RequestError exception: {str(e)}"
             )
-        return {}
-
-    def data_operation(
-        self, verb: str, operation: str, name: Optional[str], data: dict
-    ) -> dict:
-        if operation == "feedback":
-            return self.analyze(data["item"])
-        if operation == "search":
-            return self.search(data)
-        return {}
-
-    def search(self, data) -> dict:
-        try:
-            print(
-                "----------------------------------------------------------------------"
-            )
-            print(
-                "----------------------------------------------------------------------"
-            )
-            print("Running for Search")
-
-            self.output = self.analyze_parent_info(data)
 
-            try:
-                updated_output = replace_qcodes(self.output)
-                return updated_output
-            except Exception as e:
-                print(
-                    f"Error occurred in replace_qcodes while Analyzing Parent Info: {e}"
-                )
-                return self.output
-        except Exception as e:
-            print(e)
-            pass
-        return {}
-
-    def analyze(self, html_content: Dict[str, str], tags=None) -> Dict:
+    def analyze(self, html_content: str) -> dict:
```

Comment:
I'm wondering how it works with those changes, for me it doesn't run locally anymore.
also mypy complains about that type because `str` doesn't have the `items` method which is used later

Reply:
resolved.